ynput/OpenPype
# Fusion: collect variants of loaded containers to variant list for Creator

## Changelog Description
This handles specific use case where customer wants to provide artists by variant from loaded container when they are creating new product.

Loaded `renderMaxEye`  where `Eye`  is a variant >> in Variant dropdown it should populate `Eye` (at top) for artist to easily select and not to have remember if it was `Eye` , `EYE` or `EY`.

## Additional info
This PR is created from https://github.com/ynput/OpenPype/pull/6057 so it contains more modified files than it eventually will. 

This PR is set up to be a starting point of discussion. There are a couple of issues with this:
- we are not explicitly storing `variant` for published products so this PR is trying to 'guess' variant from loaded product, which might not be 100%
- is this workflow applicable/usable for multiple studios
- should this be configurable (per creator)
- could this be usable in different DCCs

## Testing notes:
1. load something in Fusion
2. try to create new product, check Variant dropdown in Publisher UI if new variant from loaded product was added
<img width="526" alt="fuscript_6iIrD8aO1W" src="https://github.com/ynput/OpenPype/assets/4457962/56263585-9d6f-4a4d-b455-de5fee13e1a4">
